src/BlockUniqidClassNameTransformer.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * Class BlockUniqidClassNameTransformer.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * Transform uniqid-based IDs into cacheable IDs based on wp_unique_id.
+ *
+ * Random strings based on `uniqid()` should not be used to generate CSS class
+ * names and used in inline styles so that parsed CSS transient caching is not
+ * automatically disabled.
+ *
+ * Instead, `wp_unique_id()` should be used so that the IDs in the class names
+ * are more predictable and the CSS transient caching works as expected.
+ *
+ * @link https://github.com/ampproject/amp-wp/pull/6925
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.2.2
+ * @internal
+ */
+final class BlockUniqidClassNameTransformer implements Conditional, Service, Registerable, Delayed {
+
+	/**
+	 * Prefixes of class names that should be transformed.
+	 *
+	 * @var string[]
+	 */
+	const CLASS_NAME_PREFIXES = [
+		'wp-container-',
+		'wp-duotone-',
+	];
```

Comment:
Note that WordPress/gutenberg#38891 covers 2 more class name prefixes:

1. `wp-block-archives-` - the `uniqid`-based class names are already sanitized and there's no need to transform them here.
2. `wp-elements-` - I couldn't find a way to render a block with such a class name so I skipped it here.

Reply:
> `wp-elements-` - I couldn't find a way to render a block with such a class name so I skipped it here.

It appears that it is limited full site editing. I haven't been able to reproduce it yet myself.

Reply:
I found a way to reproduce it. I went to the Site Editor, selected the paragraph in the footer:

> <img width="467" alt="image" src="https://user-images.githubusercontent.com/134745/157560539-57bc4ad3-a2b8-4106-a9d0-1c55cefbf5d8.png">

I went to the block's settings:

> <img width="294" alt="image" src="https://user-images.githubusercontent.com/134745/157560614-e2d7886b-f974-42b1-aa03-2402668e5c30.png">

I opened the menu and selected Link:

> <img width="289" alt="image" src="https://user-images.githubusercontent.com/134745/157560657-b7309a03-24fe-4883-a2fd-7157bf3b50cc.png">

Then I selected one of the preset colors:

> <img width="590" alt="image" src="https://user-images.githubusercontent.com/134745/157560727-1493b220-6548-4d26-b1d2-3b5d64ad0fc9.png">

Then on the frontend, the footer template part's markup is:

```html
<footer class="wp-block-template-part">
	<div class="wp-container-622941a2cc259 wp-block-group alignfull">
		<div class="wp-container-622941a2cb2af wp-block-group alignwide" style="padding-top:4rem;padding-bottom:4rem">
			<h1 class="has-text-color has-primary-color has-background has-secondary-background-color wp-block-site-title">
				<a href="https://wordpressdev.lndo.site" rel="home" aria-current="page">WordPress Develop</a>
			</h1>
			<p class="wp-elements-622941a2ca7ee has-text-align-right has-link-color">
				Proudly powered by <a href="https://wordpress.org" rel="nofollow">WordPress</a>
			</p>
		</div>
	</div>
</footer>
```

And there is a style added:

```html
<style>.wp-elements-622941a2ca7ee a{color: var(--wp--preset--color--primary);}</style>
```

This is in **Gutenberg v12.6.1**.